### PR TITLE
BL-7728: Update subscription-confirmation template with hgv/psv changes

### PR DESCRIPTION
**Pull request by mrogowska**

### CODE REVIEW

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResource.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package uk.gov.dvsa.motr.web.resource;
 
 import uk.gov.dvsa.motr.vehicledetails.MotIdentification;
+import uk.gov.dvsa.motr.vehicledetails.VehicleDetails;
```
Comment (ryszardm-kainos) [inline motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResource.java]: is this used anywhere?

File: motr-webapp/webapp/src/test/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResourceTest.java [hunk 2]
```diff
@@ -160,7 +161,36 @@ public void modelIsPopulatedWithSmsRelatedDataWhenConfirmedSignUpSms() throws Ex
...
+        SubscriptionConfirmationParams confirmationParams = new SubscriptionConfirmationParams();
+        confirmationParams.setRegistration(VRM);
+        confirmationParams.setMotTestNumber(MOT_TEST_NUMBER);
+        confirmationParams.setContactType(CONTACT_TYPE_MOBILE);
+        confirmationParams.setVehicleType(VehicleType.HGV);
+        when(motrSession.getSubscriptionConfirmationParams()).thenReturn(confirmationParams);
+
+        resource.confirmSubscriptionFirstTimeGet();
+
+        assertEquals(false, engine.getContext(Map.class).get("isMotVehicle"));
```
Comment (ryszardm-kainos) [inline motr-webapp/webapp/src/test/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResourceTest.java]: You can use `assertFalse` in this case

File: motr-webapp/webapp/src/test/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResourceTest.java [hunk 3]
```diff
@@ -160,7 +161,36 @@ public void modelIsPopulatedWithSmsRelatedDataWhenConfirmedSignUpSms() throws Ex
...
+        SubscriptionConfirmationParams confirmationParams = new SubscriptionConfirmationParams();
+        confirmationParams.setRegistration(VRM);
+        confirmationParams.setMotTestNumber(MOT_TEST_NUMBER);
+        confirmationParams.setContactType(CONTACT_TYPE_MOBILE);
+        confirmationParams.setVehicleType(VehicleType.MOT);
+        when(motrSession.getSubscriptionConfirmationParams()).thenReturn(confirmationParams);
+
+        resource.confirmSubscriptionFirstTimeGet();
 
+        assertEquals(true, engine.getContext(Map.class).get("isMotVehicle"));
```
Comment (ryszardm-kainos) [inline motr-webapp/webapp/src/test/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResourceTest.java]: `assertTrue` could be used here

File: motr-webapp/webapp/src/test/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResourceTest.java [hunk 4]
```diff
@@ -160,7 +161,36 @@ public void modelIsPopulatedWithSmsRelatedDataWhenConfirmedSignUpSms() throws Ex
...
+        assertEquals(false, engine.getContext(Map.class).get("isMotVehicle"));
+    }
+
+    @Test
+    public void isMotVehicleIsSetToTrueInModelMapWhenSubscriptionIsForMotVehicle() throws Exception {
+
+        SubscriptionConfirmationParams confirmationParams = new SubscriptionConfirmationParams();
+        confirmationParams.setRegistration(VRM);
+        confirmationParams.setMotTestNumber(MOT_TEST_NUMBER);
+        confirmationParams.setContactType(CONTACT_TYPE_MOBILE);
```
Comment (ryszardm-kainos) [inline motr-webapp/webapp/src/test/java/uk/gov/dvsa/motr/web/resource/SubscriptionConfirmedResourceTest.java]: Those 4 lines are replicated multiple times in this file. This test class could be improved nicely if we had some nice method to create a `SubscriptionConfirmationParams` instance.

File: motr-webapp/webapp/src/main/resources/template/subscription-confirmation.hbs [hunk 5]
```diff
@@ -15,7 +19,12 @@
 
             <h2 class="heading heading-medium heading--medium">What happens next?</h2>
             <p>We’ve sent you a confirmation {{#if usingSms}}text message{{else}}email{{/if}}.</p>
-            <p>You’ll get an annual reminder a month before your MOT is due, and again 2 weeks later if you don’t have your MOT yet.</p>
+            {{#if isMotVehicle}}
+                <p>You’ll get an annual reminder a month before your MOT is due, and again 2 weeks later if you don’t have your MOT yet.</p>
+            {{else}}
+                <p>You’ll get a reminder 2 months before your annual test is due, and again a month later if you don’t have your annual test yet.</p>
+                <p>You'll get a reminder every year until you unsubscribe.</p>
```
Comment (ryszardm-kainos) [inline motr-webapp/webapp/src/main/resources/template/subscription-confirmation.hbs]: Apostrophes not consistent

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/component/subscription/persistence/DynamoDbSubscriptionRepository.java [hunk 6]
```diff
@@ -157,7 +157,6 @@ private Subscription mapItemToSubscription(Item item) {
         subscription.setMotDueDate(LocalDate.parse(item.getString("mot_due_date")));
         subscription.setMotIdentification(new MotIdentification(item.getString("mot_test_number"), item.getString("dvla_id")));
         subscription.setContactDetail(contactDetail);
-        subscription.setVehicleType(VehicleType.getFromString(item.getString("vehicle_type")));
```
Comment (ryszardm-kainos) [inline motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/component/subscription/persistence/DynamoDbSubscriptionRepository.java]: 👍 can you also cleanup `vehicleType` in `save` method in this file?

Comment (mrogowska) [reply motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/component/subscription/persistence/DynamoDbSubscriptionRepository.java]: Sure thing, done.

**Pull request merged**